### ISSUE TITLE
Getting compiliation error  with the latest code

### ISSUE THREAD

**Issue by rosy8** [sysrepo/sysrepo#839]
Hi,

I am getting the following compilation error with the latest code.

[ 61%] Building C object src/CMakeFiles/SR_ENGINE.dir/rp_dt_filter.c.o
[ 64%] Building C object src/CMakeFiles/SR_ENGINE.dir/data_manager.c.o
gitcode_devel/sysrepo/src/data_manager.c: In function ‘dm_release_tmp_ly_ctx’:
<b>gitcode_devel/sysrepo/src/data_manager.c:567:20: error: ‘LY_INTERNAL_MODULE_COUNT’ undeclared (first use in this function) </b>
     uint32_t idx = LY_INTERNAL_MODULE_COUNT;
                    ^
gitcode_devel/sysrepo/src/data_manager.c:567:20: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [src/CMakeFiles/SR_ENGINE.dir/data_manager.c.o] Error 1
make[1]: *** [src/CMakeFiles

I just made the below changes 
//uint32_t idx = LY_INTERNAL_MODULE_COUNT;
 uint32_t idx = 5;  (As specified in side libyang.h file    #define LY_INTERNAL_MODULE_COUNT 5 )

With this compilation was successful but getting the following error during installation..
[root@18-88 lib64]# /usr/bin/sysrepoctl --install --yang=/etc/libnetconf/yang-models/ietf-inet-types@2013-07-15.yang
Installing a new module from file '/etc/libnetconf/yang-models/ietf-inet-types@2013-07-15.yang'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-inet-types@2013-07-15.yang'...
<b>[ERR] (srctl_ly_log_cb:487) libyang: Schema node not found.</b>
[ERR] (md_lyd_new_path:413) Failed to set installed flag for module 'ietf-inet-types@2013-07-15': Schema node not found.
Error: Unable to insert the module into the dependency graph.
Reverting the install operation...
Deleted the schema file '/etc/sysrepo/yang/ietf-inet-types@2013-07-15.yang'.
Install operation failed.
[root@18-88 lib64]#

Please let me know how to overcome this issue.
Thanks in advance.

Regards,
Rosy

**Comment by michalvasko** [comment]
Hi Rosy,
no modifications are necessary, just update *libyang*.

Regards,
Michal

**Comment by rosy8** [comment]
Hi Michal,

All (sysrepo/netopeer2/libnetconf2/libyang) are latest code.
But still facing this issue.

Please let me know if i am missing something here or i need to use any other command for installation.

Regards,
Rosy

**Comment by michalvasko** [comment]
Hi Rosy,
just look here cesnet/libyang@2d051a137cda1850e91d4c4, the macro was added to a public header, so if you `# make install` a recent *libyang*, *sysrepo* should find it. You can try to uninstall all *libyang* installations you find on your system and then reinstall the newest one, simply make sure the newest header files and library is used.

Regards,
Michal

**Comment by rosy8** [comment]
Hi Michal,

I donot see any  "uninstall' command in Makefile. Would you please let me know how to do that?

Instead what I did is that, I removed the "build" dir  from libyang and sysrepo folders.
And again followed the same steps of compilation, with that I am getting the following error

[ 98%] Building CXX object swig/lua/CMakeFiles/Sysrepo_Lua.dir/__/cpp/src/Tree.cpp.o
[ 99%] Building CXX object swig/lua/CMakeFiles/Sysrepo_Lua.dir/__/cpp/src/Internal.cpp.o
Linking CXX shared library libSysrepo_Lua.so
[ 99%] Built target Sysrepo_Lua
[100%] Swig source
/gitcode_devel/sysrepo/swig/cpp/src/Sysrepo.h:107: Error: Syntax error in input(3).
make[2]: *** [swig/lua/libsysrepoLuaLUA_wrap.cxx] Error 1
make[1]: *** [swig/lua/CMakeFiles/libsysrepoLua.dir/all] Error 2
make: *** [all] Error 2
-sh-4.2$ make

Regards,
Rosy

**Comment by michalvasko** [comment]
Hi Rosy,
you have to perform uninstall manually, remove all `libyang.so` and `libyang.h` from your system directories.

Regarding C++ bindings, I cannot help you, maybe someone else can. You can perhaps try updating `swig`.

Regards,
Michal

**Comment by mislavn** [comment]
Hi Rosy,

The C++ error has known to occur when the build directory is not properly cleaned or when the github repo has been updated and an old build directory has been used.

This is fixed with the latest master branch, to fix it now you will need to remove everything from the current build directory `cd build; rm -rf *;`.

Regards,
Mislav

**Comment by rosy8** [comment]
Hi Michal,

Compilation is working fine now . I donot need to manually change the data_manager.c file to resolve the compilation error related to "LY_INTERNAL_MODULE_COUNT’.

 But still I am getting this error while installing..
[root@18-88 ~]# /usr/bin/sysrepoctl --install --yang=/etc/libnetconf/yang-models/ietf-inet-types@2013-07-15.yang
Installing a new module from file '/etc/libnetconf/yang-models/ietf-inet-types@2013-07-15.yang'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-inet-types@2013-07-15.yang'...
</b>[ERR] (srctl_ly_log_cb:487) libyang: Schema node not found.
[ERR] (md_lyd_new_path:413) Failed to set installed flag for module 'ietf-inet-types@2013-07-15': Schema node not found. </b>
<b>Error: Unable to insert the module into the dependency graph. </b>
Reverting the install operation...
Deleted the schema file '/etc/sysrepo/yang/ietf-inet-types@2013-07-15.yang'.
Install operation failed.
[root@18-88 ~]#

Regards,
Rosy

**Comment by michalvasko** [comment]
Hi Rosy,
why exactly are you trying to install `ietf-yang-types`? It is a part of every context and anyway, it is always used just as an import, no need to directly install it. But the error is strange, I would expect something different.

Regards,
Michal

**Comment by rosy8** [comment]
Hi Michal,

This is juts an example. 
For any yang file, I am getting such type of error.

Regards,
Rosy

**Comment by michalvasko** [comment]
Yeah, sorry, I edited my comment and now it should be more helpful.

Regards,
Michal

**Comment by vishalwc** [comment]
Hi Michal,

after removing all the file from /etc/sysrepo and tried with libraries generated with latest  devel still the same problem exist

**Comment by michalvasko** [comment]
Hi,
I don't know where your repository is located (default is `sysrepo/build/repository`), but you should remove it and then try again. Does `# make install` succeed?

Regards,
Michal

**Comment by vishalwc** [comment]
yes install was successful . cleaned all /etc/sysrepo/ and sysrepo/build/repository 
but still getting the issue

**Comment by michalvasko** [comment]
Hi,
so if I understand it correctly, you clean your repository, then install *sysrepo*, and then try to install another module, which fails? If so, what model is it? I need to reproduce this to help you.

Regards,
Michal

**Comment by vishalwc** [comment]
/usr/bin/sysrepoctl -L 4 --install --yang=/ovn/etc/libnetconf/yang-models/ietf-interfaces@2014-05-08.yang
[DBG] Setting log level for stderr logs to 4.
Installing a new module from file '/ovn/etc/libnetconf/yang-models/ietf-interfaces@2014-05-08.yang'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-interfaces@2014-05-08.yang'...
[ERR] libyang: Schema node not found.
[ERR] Failed to set installed flag for module 'ietf-interfaces@2014-05-08': Schema node not found.
Error: Unable to insert the module into the dependency graph.
Reverting the install operation...
Deleted the schema file '/etc/sysrepo/yang/ietf-interfaces@2014-05-08.yang'.
Install operation failed.

**Comment by michalvasko** [comment]
Hi,
could you please also post the output of `sysrepoctl -l` *before* trying to install this new model?

Regards,
Michal

**Comment by vishalwc** [comment]
No Module  is getting installed 

Sysrepo schema directory: /etc/sysrepo/yang/
Sysrepo data directory:   /etc/sysrepo/data/
(Do not alter contents of these directories manually)

Module Name                   | Revision   | Conformance | Data Owner          | Permissions | Submodules                    | Enabled Features
-----------------------------------------------------------------------------------------------------------------------------------------------

**Comment by michalvasko** [comment]
Okay, then try the following steps, I am assuming your repository location (`REPOSITORY_LOC` compile variable) is `/etc/sysrepo` and you are building *sysrepo* in a separate `build` directory. Remove all the contents of `/etc/sysrepo`, make sure it is empty. Then do the same with the `build` directory. The problem must be some old files (specifically, `sysrepo-module-dependencies.yang` you install is of an older revision), so once you properly clean everything it will work. I can hardly help you more at this point.

Regards,
Michal

**Comment by vishalwc** [comment]
Thanks Michal it is working now sysrepod and netopeer comes up 

However we get some old error ( we are on Latest devel ) 
May 10 22:29:17 MaestrOS sysrepod[8845]: [ERR] (dm_ly_log_cb:1326) libyang error: Leafref "/ietf-keystore:keystore/ietf-keystore:p                                                                        rivate-keys/ietf-keystore:private-key/ietf-keystore:name" of value "ssh_host_rsa_key" points to a non-existing leaf.
May 10 22:29:17 MaestrOS sysrepod[8845]: [ERR] (rp_dt_commit:819) Data validation failed: Leafref "/ietf-keystore:keystore/ietf-ke                                                                        ystore:private-keys/ietf-keystore:private-key/ietf-keystore:name" of value "ssh_host_rsa_key" points to a non-existing leaf.

**Comment by michalvasko** [comment]
Hi,
that will likely be caused by you doing something during installation that you should not have. If you do not want server configuration, you should just install *sysrepo*, then *netopeer2-server*. When installing the server, it should detect that there is no `ietf-keystore` model in *sysrepo*, so it skips installing all the configuration-related stuff including its default configuration, which is causing your problem. If you want configuration, you must install *keystored* before installing the server, which actually adds the missing leafrefs. All of this is written in `netopeer2/server/configuration/README.md`. Either way, the result should not be the error you are having.

Regards,
Michal

**Comment by vishalwc** [comment]
we are installing the ietf-keytore along with all the modules and configuring it, subsequently starts the daemon as follows 
sysrepod
netopeer2-server 
client app to connect sysrepod

**Comment by michalvasko** [comment]
Hi,
I have no chance of helping you with the information you provided. So, please, *exactly* follow my instructions:

- uninstall all the applications to the extent you can, mainly *sysrepo* repository, the rest can probably stay (*libyang* and *libnetconf2* can be left unchanged),
- install *sysrepo*,
- install *keystored* by first removing `CMakeCache.txt`, then executing `cmake`, and posting its output here,
- install *netopeer2-server*, again first removing `CMakeCache.txt`, executing `cmake`, and putting the output here.

Regards,
Michal

**Comment by vishalwc** [comment]
Hi Michal ,
We clean and install the lib as per your steps  however posting  the logs for your reference 
The issue is still reproducible 

**MaestrOS sysrepod[8845]: [ERR] (dm_ly_log_cb:1326) libyang error: Leafref "/ietf-keystore:keystore/ietf-keystore:p rivate-keys/ietf-keystore:private-key/ietf-keystore:name" of value "ssh_host_rsa_key" points to a non-existing leaf.**
**MaestrOS sysrepod[8845]: [ERR] (rp_dt_commit:819) Data validation failed: Leafref "/ietf-keystore:keystore/ietf-ke ystore:private-keys/ietf-keystore:private-key/ietf-keystore:name" of value "ssh_host_rsa_key" points to a non-existing leaf.**




**Following are the sysrepo install logs**

[BEGIN] 23-May-17 12:24:09 PM
 .. && make && sudo make install
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/lib64/ccache/cc
-- Check for working C compiler: /usr/lib64/ccache/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/lib64/ccache/c++
-- Check for working CXX compiler: /usr/lib64/ccache/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Preparing release build of sysrepo v. 0.6.1
-- sysrepo repository location: /etc/sysrepo
-- Found ev: /usr/lib64/libev.so  
-- Found yang: /usr/local/lib64/libyang.so  
-- Found protobuf-c: /usr/lib64/libprotobuf-c.so  
-- Could NOT find avl (missing:  AVL_LIBRARY AVL_INCLUDE_DIR) 
-- Found redblack: /usr/local/lib/libredblack.so  
-- libredblack will be used for binary tree manipulations.
-- Looking for pthread_rwlockattr_setkind_np
-- Looking for pthread_rwlockattr_setkind_np - found
-- Looking for getpeereid
-- Looking for getpeereid - not found
-- Looking for getpeerucred
-- Looking for getpeerucred - not found
-- Looking for include file ucred.h
-- Looking for include file ucred.h - not found
-- Looking for pthread_mutex_timedlock
-- Looking for pthread_mutex_timedlock - found
-- Looking for setfsuid
-- Looking for setfsuid - found
-- Looking for fsetxattr
-- Looking for fsetxattr - found
-- Performing Test HAVE_STAT_ST_MTIM
-- Performing Test HAVE_STAT_ST_MTIM - Success
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.5") 
-- Configuring done
-- Generating done
-- Build files have been written to: .//sysrepo/build
[  2%] Running C protocol buffer compiler on .//sysrepo/src/sysrepo.proto
Scanning dependencies of target COMMON
[  5%] Building C object src/CMakeFiles/COMMON.dir/common/sysrepo.pb-c.c.o
[  8%] Building C object src/CMakeFiles/COMMON.dir/common/sr_common.c.o
[ 11%] Building C object src/CMakeFiles/COMMON.dir/common/sr_utils.c.o
[ 14%] Building C object src/CMakeFiles/COMMON.dir/common/sr_data_structs.c.o
[ 17%] Building C object src/CMakeFiles/COMMON.dir/common/sr_logger.c.o
[ 20%] Building C object src/CMakeFiles/COMMON.dir/common/sr_protobuf.c.o
[ 23%] Building C object src/CMakeFiles/COMMON.dir/common/sr_mem_mgmt.c.o
[ 26%] Building C object src/CMakeFiles/COMMON.dir/utils/plugins.c.o
[ 29%] Building C object src/CMakeFiles/COMMON.dir/utils/trees.c.o
[ 32%] Building C object src/CMakeFiles/COMMON.dir/utils/values.c.o
[ 35%] Building C object src/CMakeFiles/COMMON.dir/utils/xpath.c.o
[ 35%] Built target COMMON
Scanning dependencies of target SR_ENGINE
[ 38%] Building C object src/CMakeFiles/SR_ENGINE.dir/access_control.c.o
[ 41%] Building C object src/CMakeFiles/SR_ENGINE.dir/connection_manager.c.o
[ 44%] Building C object src/CMakeFiles/SR_ENGINE.dir/cm_session_manager.c.o
[ 47%] Building C object src/CMakeFiles/SR_ENGINE.dir/request_processor.c.o
[ 50%] Building C object src/CMakeFiles/SR_ENGINE.dir/rp_dt_xpath.c.o
[ 52%] Building C object src/CMakeFiles/SR_ENGINE.dir/rp_dt_lookup.c.o
[ 55%] Building C object src/CMakeFiles/SR_ENGINE.dir/rp_dt_get.c.o
[ 58%] Building C object src/CMakeFiles/SR_ENGINE.dir/rp_dt_edit.c.o
[ 61%] Building C object src/CMakeFiles/SR_ENGINE.dir/rp_dt_filter.c.o
[ 64%] Building C object src/CMakeFiles/SR_ENGINE.dir/data_manager.c.o
[ 67%] Building C object src/CMakeFiles/SR_ENGINE.dir/notification_processor.c.o
[ 70%] Building C object src/CMakeFiles/SR_ENGINE.dir/persistence_manager.c.o
[ 73%] Building C object src/CMakeFiles/SR_ENGINE.dir/module_dependencies.c.o
[ 76%] Building C object src/CMakeFiles/SR_ENGINE.dir/nacm.c.o
[ 76%] Built target SR_ENGINE
Scanning dependencies of target SR_SRC
[ 79%] Building C object src/CMakeFiles/SR_SRC.dir/clientlib/client_library.c.o
[ 82%] Building C object src/CMakeFiles/SR_SRC.dir/clientlib/cl_subscription_manager.c.o
[ 85%] Building C object src/CMakeFiles/SR_SRC.dir/clientlib/cl_common.c.o
[ 88%] Building C object src/CMakeFiles/SR_SRC.dir/clientlib/cl_trees.c.o
[ 88%] Built target SR_SRC
Scanning dependencies of target sysrepo
Linking C shared library libsysrepo.so
[ 88%] Built target sysrepo
Scanning dependencies of target sysrepo-plugind
[ 91%] Building C object src/CMakeFiles/sysrepo-plugind.dir/executables/sysrepo-plugind.c.o
Linking C executable sysrepo-plugind
[ 91%] Built target sysrepo-plugind
Scanning dependencies of target sysrepo_a
Linking C static library libsysrepo.a
[ 91%] Built target sysrepo_a
Scanning dependencies of target sysrepocfg
[ 94%] Building C object src/CMakeFiles/sysrepocfg.dir/executables/sysrepocfg.c.o
Linking C executable sysrepocfg
[ 94%] Built target sysrepocfg
Scanning dependencies of target sysrepoctl
[ 97%] Building C object src/CMakeFiles/sysrepoctl.dir/executables/sysrepoctl.c.o
Linking C executable sysrepoctl
[ 97%] Built target sysrepoctl
Scanning dependencies of target sysrepod
[100%] Building C object src/CMakeFiles/sysrepod.dir/executables/sysrepod.c.o
Linking C executable sysrepod
[100%] Built target sysrepod
[ 35%] Built target COMMON
[ 76%] Built target SR_ENGINE
[ 88%] Built target SR_SRC
[ 88%] Built target sysrepo
[ 91%] Built target sysrepo-plugind
[ 91%] Built target sysrepo_a
[ 94%] Built target sysrepocfg
[ 97%] Built target sysrepoctl
[100%] Built target sysrepod
Install the project...
-- Install configuration: "Release"
-- Installing: /etc/sysrepo
-- Installing: /etc/sysrepo/yang/
-- Installing: /etc/sysrepo/data/
-- Installing: /etc/sysrepo/data/internal/
-- Installing: /var/run/sysrepo-subscriptions
-- Installing: /usr/local/lib64/sysrepo/plugins
-- Up-to-date: /etc/sysrepo/yang/internal/sysrepo-persistent-data.yang
-- Up-to-date: /etc/sysrepo/yang/internal/sysrepo-module-dependencies.yang
-- Up-to-date: /etc/sysrepo/yang/internal/sysrepo-notification-store.yang
-- Exec: .//sysrepo/build/src/sysrepoctl --install --yang=.//sysrepo/yang/ietf-netconf-acm@2012-02-22.yang --permissions=644
-- Installing: /usr/local/lib64/pkgconfig/libsysrepo.pc
-- Up-to-date: /usr/local/share/man/man1/sysrepoctl.1
-- Up-to-date: /usr/local/share/man/man1/sysrepocfg.1
-- Exec: .//sysrepo/build/src/sysrepoctl --install --yang=.//sysrepo/yang/ietf-netconf-notifications.yang --permissions=666
-- Exec: .//sysrepo/build/src/sysrepoctl --install --yang=.//sysrepo/yang/nc-notifications.yang --permissions=666
-- Exec: .//sysrepo/build/src/sysrepoctl --install --yang=.//sysrepo/yang/notifications.yang --permissions=666
-- Installing: /usr/local/lib64/libsysrepo.so.0.6.1
-- Installing: /usr/local/lib64/libsysrepo.so.0.6
-- Installing: /usr/local/lib64/libsysrepo.so
-- Removed runtime path from "/usr/local/lib64/libsysrepo.so.0.6.1"
-- Installing: /usr/local/lib64/libsysrepo.a
-- Installing: /usr/local/bin/sysrepod
-- Removed runtime path from "/usr/local/bin/sysrepod"
-- Installing: /usr/local/bin/sysrepo-plugind
-- Removed runtime path from "/usr/local/bin/sysrepo-plugind"
-- Installing: /usr/local/bin/sysrepoctl
-- Removed runtime path from "/usr/local/bin/sysrepoctl"
-- Installing: /usr/local/bin/sysrepocfg
-- Removed runtime path from "/usr/local/bin/sysrepocfg"
-- Installing: /usr/local/include
-- Up-to-date: /usr/local/include/sysrepo.h
-- Installing: /usr/local/include/sysrepo
-- Up-to-date: /usr/local/include/sysrepo/plugins.h
-- Up-to-date: /usr/local/include/sysrepo/trees.h
-- Up-to-date: /usr/local/include/sysrepo/values.h
-- Up-to-date: /usr/local/include/sysrepo/xpath.h
-bash-4.2$ 





Floowing are the keystored install logs:--
-----------------------------------------


-- The C compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/lib64/ccache/cc
-- Check for working C compiler: /usr/lib64/ccache/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.27.1") 
CMake Warning at CMakeLists.txt:57 (message):
  keystored pkg-config file copied into a directory not searched by
  pkg-config.


CMake Warning at CMakeLists.txt:58 (message):
  For netopeer2-server configuration to work, pkg-config search path must
  include "/usr/local/lib64/pkgconfig" or keystored keys directory will have
  to be set manually to "/usr/local/etc/keystored/keys".


-- Found sysrepo: /usr/local/lib64/libsysrepo.so  
-- Configuring done
-- Generating done
-- Build files have been written to: .//Netopeer2/keystored/build
Scanning dependencies of target keystored
[100%] Building C object CMakeFiles/keystored.dir/keystored.c.o
.//Netopeer2/keystored/keystored.c: In function ‘ks_privkey_gen_cb’:
.//Netopeer2/keystored/keystored.c:414:5: warning: ‘ret’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     return ret;
     ^
.//Netopeer2/keystored/keystored.c: In function ‘sr_plugin_init_cb’:
.//Netopeer2/keystored/keystored.c:538:9: warning: ‘rc’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     int rc;
         ^
Linking C shared library libkeystored.so
[100%] Built target keystored
[100%] Built target keystored
Install the project...
-- Install configuration: "Release"
-- Installing: /usr/local/lib64/pkgconfig/keystored.pc
-- Installing: /usr/local/etc/keystored/keys
-- Module ietf-keystore already in sysrepo.
-- Some ietf-keystore configuration set, no keys will be imported.
-- Installing: /usr/local/lib64/sysrepo/plugins/libkeystored.so
-- Removed runtime path from "/usr/local/lib64/sysrepo/plugins/libkeystored.so"




Following are the netopeer2 installed logs
-------------------------------------------

-bash-4.2$ cd .//Netopeer2/server/build && rm -rf * && cmake -D CMAKE_BUILD_TYPE:String="Release" .. && make && sudo make install
-- The C compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/lib64/ccache/cc
-- Check for working C compiler: /usr/lib64/ccache/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.27.1") 
-- Failed to learn keystored keys directory, rerun cmake and set KEYSTORED_KEYS_DIR manually or the server configuration will not work.
-- Server configuration off, default SSH host key will be used.
-- Default host key set to "/etc/ssh/ssh_host_rsa_key".
-- Unable to learn libnetconf2 thread support, check skipped
-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Looking for pthread_rwlockattr_setkind_np
-- Looking for pthread_rwlockattr_setkind_np - found
-- Looking for NC_ENABLED_SSH
-- Looking for NC_ENABLED_SSH - found
-- Looking for NC_ENABLED_TLS
-- Looking for NC_ENABLED_TLS - found
-- Found sysrepo: /usr/local/lib64/libsysrepo.so  
-- Configuring done
-- Generating done
-- Build files have been written to: .//Netopeer2/server/build
Scanning dependencies of target serverobj
[  6%] Building C object CMakeFiles/serverobj.dir/ietf_netconf_server.c.o
[ 12%] Building C object CMakeFiles/serverobj.dir/ietf_system.c.o
[ 18%] Building C object CMakeFiles/serverobj.dir/ietf_keystore.c.o
[ 25%] Building C object CMakeFiles/serverobj.dir/netconf_monitoring.c.o
[ 31%] Building C object CMakeFiles/serverobj.dir/operations.c.o
[ 37%] Building C object CMakeFiles/serverobj.dir/op_get_config.c.o
[ 43%] Building C object CMakeFiles/serverobj.dir/op_editconfig.c.o
[ 50%] Building C object CMakeFiles/serverobj.dir/op_copyconfig.c.o
[ 56%] Building C object CMakeFiles/serverobj.dir/op_deleteconfig.c.o
[ 62%] Building C object CMakeFiles/serverobj.dir/op_candidate.c.o
[ 68%] Building C object CMakeFiles/serverobj.dir/op_validate.c.o
[ 75%] Building C object CMakeFiles/serverobj.dir/op_un_lock.c.o
[ 81%] Building C object CMakeFiles/serverobj.dir/op_generic.c.o
[ 87%] Building C object CMakeFiles/serverobj.dir/op_notifications.c.o
In file included from .//Netopeer2/server/common.h:21:0,
                 from .//Netopeer2/server/op_notifications.c:29:
.//Netopeer2/server/op_notifications.c: In function ‘np2srv_ntf_clb’:
.//Netopeer2/server/log.h:49:81: warning: ‘ntf_type_str’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 #define VRB(format,args...) if(np2_verbose_level>=NC_VERB_VERBOSE){np2log_printf(NC_VERB_VERBOSE,format,##args);}
                                                                                 ^
.//Netopeer2/server/op_notifications.c:477:17: note: ‘ntf_type_str’ was declared here
     const char *ntf_type_str, *module_name = NULL;
                 ^
[ 93%] Building C object CMakeFiles/serverobj.dir/log.c.o
[ 93%] Built target serverobj
Scanning dependencies of target netopeer2-server
[100%] Building C object CMakeFiles/netopeer2-server.dir/main.c.o
Linking C executable netopeer2-server
[100%] Built target netopeer2-server
[ 93%] Built target serverobj
[100%] Built target netopeer2-server
Install the project...
-- Install configuration: "Release"
-- Installing: /usr/local/bin/netopeer2-server
-- Removed runtime path from "/usr/local/bin/netopeer2-server"
-- Module ietf-netconf-server already in sysrepo.
-- Module ietf-system already in sysrepo.
-- Some NETCONF server configuration set, it will not be changed.

[END] 23-May-17 12:33:38 PM

**Comment by michalvasko** [comment]
Hi,
from the log I can see that server configuration was not installed because of the reasons mentioned in the logs. However, the data (for `ietf-netconf-server`) were obviously installed before (otherwise you would not get the error) so it is likely you have `ietf-keystore` data, too. In that case, just start *sysrepo-plugind*, which must be running with *keystored* for that missing data to be available (also written in `server/configuration/README.md`, which you should have read).

Regards,
Michal

**Comment by vishalwc** [comment]
Here is the sequence 
- Install the sysrepod ,sysrepocfg ,sysrepoctl, netopeer2-server , sysrepo-plugin and libkeystored  
- Install all the models including ietf-netconf-server ,ietf-keystore
-  **using sysrepocfg configure ietf-keystore and ietf-netconf-server ( this for 830 default port and enable callhome)  thsi give and error.**

start sysrepod , sysrepo-plugin,netopeer2-server

what should be done to get the keys install and which allows to configure ietf-netconf-server

**Comment by michalvasko** [comment]
Hi,
you were doing all of those points manually? All you need to do is `# make install` in first *sysrepo*, then *keystored*, and lastly *netopeer2-server* (the order is important). It will also install all models and the default configuration, which you can then modify.

Regards,
Michal

**Comment by vishalwc** [comment]
Hi Michal ,

we did not change our device installation it is working till 11th April build and  as I mentioned even in cmake install it is the same , we install in the same way 

/usr/bin/sysrepocfg -d startup -i /home/vishalc/Netconf_v2/Netopeer2/keystored/stock_key_config.xml ietf-keystore 
**the above is installed I verified in the ietf-keystore.startup** 

**below give as error in our installaion **
/usr/bin/sysrepocfg -d startup -i /home/vishalc/Netconf_v2/Netopeer2/server/stock_config.xml ietf-netconf-server

please see your old fix  since than the installation had not changed however the issue is resurfacing .
https://github.com/CESNET/Netopeer2/issues/130

please let me know if you need same info to install the file

**Comment by michalvasko** [comment]
Hi,
we better start over because there were a lot of changes made since that issue, so please post the output of `# make install` on *keystored* and *netopeer2-server*.

Regards,
Michal

**Comment by vishalwc** [comment]
Hi Michal 
Thanks for you support 
we do not use make install on device , however we have done the same step of netopeer  cmake install in our script of device for installation

**Comment by michalvasko** [comment]
Right, you could have mentioned this before. Then I guess post the whole output of your custom install commands, with the commands themselves please.

Regards,
Michal

**Comment by vishalwc** [comment]
Installing a new module from file '/ovn/etc/libnetconf/yang-models/ietf-inet-types@2013-07-15.yang'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-inet-types@2013-07-15.yang'...
Skipping installation of data files for module 'ietf-inet-types'...
Notifying sysrepo about the change...
Install operation completed successfully.
Installing a new module from file '/ovn/etc/libnetconf/yang-models/ietf-yang-types@2013-07-15.yang'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-yang-types@2013-07-15.yang'...
Skipping installation of data files for module 'ietf-yang-types'...
Notifying sysrepo about the change...
Install operation completed successfully.
Installing a new module from file '/ovn/etc/libnetconf/yang-models/ietf-netconf-monitoring@2010-10-04.yang'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-netconf-monitoring@2010-10-04.yang'...
Installing data files for module 'ietf-netconf-monitoring'...
Notifying sysrepo about the change...
Install operation completed successfully.
Installing a new module from file '/ovn/etc/libnetconf/yang-models/ietf-netconf-acm@2012-02-22.yang'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-netconf-acm@2012-02-22.yang'...
Installing data files for module 'ietf-netconf-acm'...
Notifying sysrepo about the change...
Install operation completed successfully.
Installing a new module from file '/ovn/etc/libnetconf/yang-models/ietf-netconf@2011-06-01.yang'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-netconf@2011-06-01.yang'...
Resolving dependency: 'ietf-netconf' imports 'ietf-netconf-acm'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-netconf-acm@2012-02-22.yang'...
Installing data files for module 'ietf-netconf'...
Resolving dependency: 'ietf-netconf' imports 'ietf-netconf-acm'...
Installing data files for module 'ietf-netconf-acm'...
Notifying sysrepo about the change...
Install operation completed successfully.
Installing a new module from file '/ovn/etc/libnetconf/yang-models/ietf-interfaces@2014-05-08.yang'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-interfaces@2014-05-08.yang'...
Installing data files for module 'ietf-interfaces'...
Notifying sysrepo about the change...
Install operation completed successfully.
Installing a new module from file '/ovn/etc/libnetconf/yang-models/iana-if-type.yang'...
Installing the YANG file to '/etc/sysrepo/yang/iana-if-type@2014-05-08.yang'...
Resolving dependency: 'iana-if-type' imports 'ietf-interfaces'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-interfaces@2014-05-08.yang'...
Skipping installation of data files for module 'iana-if-type'...
Resolving dependency: 'iana-if-type' imports 'ietf-interfaces'...
Installing data files for module 'ietf-interfaces'...
Notifying sysrepo about the change...
Install operation completed successfully.
Installing a new module from file '/ovn/etc/libnetconf/yang-models/ietf-netconf-with-defaults.yang'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-netconf-with-defaults@2011-06-01.yang'...
Resolving dependency: 'ietf-netconf-with-defaults' imports 'ietf-netconf'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-netconf@2011-06-01.yang'...
Resolving dependency: 'ietf-netconf' imports 'ietf-netconf-acm'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-netconf-acm@2012-02-22.yang'...
Skipping installation of data files for module 'ietf-netconf-with-defaults'...
Resolving dependency: 'ietf-netconf-with-defaults' imports 'ietf-netconf'...
Installing data files for module 'ietf-netconf'...
Resolving dependency: 'ietf-netconf' imports 'ietf-netconf-acm'...
Installing data files for module 'ietf-netconf-acm'...
Notifying sysrepo about the change...
Install operation completed successfully.
Installing a new module from file '/ovn/etc/libnetconf/yang-models/ietf-system@2014-08-06.yang'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-system@2014-08-06.yang'...
Resolving dependency: 'ietf-system' imports 'ietf-netconf-acm'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-netconf-acm@2012-02-22.yang'...
Resolving dependency: 'ietf-system' imports 'iana-crypt-hash'...
Installing the YANG file to '/etc/sysrepo/yang/iana-crypt-hash@2014-08-06.yang'...
Installing data files for module 'ietf-system'...
Resolving dependency: 'ietf-system' imports 'ietf-netconf-acm'...
Installing data files for module 'ietf-netconf-acm'...
Resolving dependency: 'ietf-system' imports 'iana-crypt-hash'...
Skipping installation of data files for module 'iana-crypt-hash'...
Notifying sysrepo about the change...
Install operation completed successfully.
Installing a new module from file '/ovn/etc/libnetconf/yang-models/iana-crypt-hash@2014-08-06.yang'...
Installing the YANG file to '/etc/sysrepo/yang/iana-crypt-hash@2014-08-06.yang'...
Skipping installation of data files for module 'iana-crypt-hash'...
Notifying sysrepo about the change...
Install operation completed successfully.
Installing a new module from file '/ovn/etc/libnetconf/yang-models/ietf-netconf-server@2016-11-02.yang'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-netconf-server@2016-11-02.yang'...
Resolving dependency: 'ietf-netconf-server' imports 'ietf-x509-cert-to-name'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-x509-cert-to-name@2014-12-10.yang'...
Resolving dependency: 'ietf-netconf-server' imports 'ietf-ssh-server'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-ssh-server@2016-11-02.yang'...
Resolving dependency: 'ietf-ssh-server' imports 'ietf-keystore'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-keystore@2016-10-31.yang'...
Resolving dependency: 'ietf-netconf-server' imports 'ietf-tls-server'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-tls-server@2016-11-02.yang'...
Resolving dependency: 'ietf-tls-server' imports 'ietf-keystore'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-keystore@2016-10-31.yang'...
Installing data files for module 'ietf-netconf-server'...
Resolving dependency: 'ietf-netconf-server' imports 'ietf-x509-cert-to-name'...
Skipping installation of data files for module 'ietf-x509-cert-to-name'...
Resolving dependency: 'ietf-netconf-server' imports 'ietf-ssh-server'...
Skipping installation of data files for module 'ietf-ssh-server'...
Resolving dependency: 'ietf-ssh-server' imports 'ietf-keystore'...
Installing data files for module 'ietf-keystore'...
Resolving dependency: 'ietf-netconf-server' imports 'ietf-tls-server'...
Skipping installation of data files for module 'ietf-tls-server'...
Resolving dependency: 'ietf-tls-server' imports 'ietf-keystore'...
Installing data files for module 'ietf-keystore'...
Notifying sysrepo about the change...
Install operation completed successfully.
Installing a new module from file '/ovn/etc/libnetconf/yang-models/ietf-netconf-notifications@2012-02-06.yang'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-netconf-notifications@2012-02-06.yang'...
Resolving dependency: 'ietf-netconf-notifications' imports 'ietf-netconf'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-netconf@2011-06-01.yang'...
Resolving dependency: 'ietf-netconf' imports 'ietf-netconf-acm'...
Installing the YANG file to '/etc/sysrepo/yang/ietf-netconf-acm@2012-02-22.yang'...
Installing data files for module 'ietf-netconf-notifications'...
Resolving dependency: 'ietf-netconf-notifications' imports 'ietf-netconf'...
Installing data files for module 'ietf-netconf'...
Resolving dependency: 'ietf-netconf' imports 'ietf-netconf-acm'...
Installing data files for module 'ietf-netconf-acm'...
Notifying sysrepo about the change...
Install operation completed successfully.
Installing a new module from file '/ovn/etc/libnetconf/yang-models/nc-notifications@2008-07-14.yang'...
Installing the YANG file to '/etc/sysrepo/yang/nc-notifications@2008-07-14.yang'...
Resolving dependency: 'nc-notifications' imports 'notifications'...
Installing the YANG file to '/etc/sysrepo/yang/notifications@2008-07-14.yang'...
Installing data files for module 'nc-notifications'...
Resolving dependency: 'nc-notifications' imports 'notifications'...
Installing data files for module 'notifications'...
Notifying sysrepo about the change...
Install operation completed successfully.
Enable Call-home feature
Enabling feature 'listen' in the module 'ietf-netconf-server'.
Operation completed successfully.
Enabling feature 'ssh-listen' in the module 'ietf-netconf-server'.
Operation completed successfully.
Enabling feature 'call-home' in the module 'ietf-netconf-server'.
Operation completed successfully.
Enabling feature 'ssh-call-home' in the module 'ietf-netconf-server'.
Operation completed successfully.
Enabling feature 'if-mib' in the module 'ietf-interfaces'.
Operation completed successfully.
writing RSA key
[WRN] Sysrepo daemon not detected. Connecting to local Sysrepo Engine.
[INF] Local Sysrepo Engine not running yet, initializing new one.
[INF] Initializing Data Manager, schema_search_dir=/etc/sysrepo/yang/, data_search_dir=/etc/sysrepo/data/
[INF] Sysrepo is running in the local mode => NACM will be disabled.
[INF] Connected to local Sysrepo Engine at socket=/tmp/sysrepo-local-19779.sock
[INF] New session created successfully, real user=root, effective user=(null), session id=341769122.
[INF] Get items request startup datastore, xpath: /ietf-keystore:*//., offset: 0, limit: 100
[INF] Data file /etc/sysrepo/data/ietf-keystore.startup is empty
[INF] Get items request startup datastore, xpath: /ietf-keystore:*//., offset: 3, limit: 100
[INF] Set item request startup datastore, xpath: /ietf-keystore:keystore/private-keys/private-key[name='ssh_host_rsa_key']
[INF] Data file /etc/sysrepo/data/ietf-keystore.startup is empty
[INF] Unlock datastore request
The new configuration was successfully applied.
[INF] Unlock datastore request
[INF] Dropping session id=341769122.
[INF] Connection Manager stop requested.
[INF] Data file /etc/sysrepo/data/ietf-netconf-notifications.running is empty
[INF] Unlock datastore request
[INF] Connection Manager successfully destroyed.
***\nFinished installing keystore\n**
[WRN] Sysrepo daemon not detected. Connecting to local Sysrepo Engine.
[INF] Local Sysrepo Engine not running yet, initializing new one.
[INF] Initializing Data Manager, schema_search_dir=/etc/sysrepo/yang/, data_search_dir=/etc/sysrepo/data/
[INF] Sysrepo is running in the local mode => NACM will be disabled.
[INF] Connected to local Sysrepo Engine at socket=/tmp/sysrepo-local-19785.sock
[INF] New session created successfully, real user=root, effective user=(null), session id=341769122.
[INF] Get items request startup datastore, xpath: /ietf-netconf-server:*//., offset: 0, limit: 100
[INF] Data file /etc/sysrepo/data/ietf-netconf-server.startup is empty
[INF] Set item request startup datastore, xpath: /ietf-netconf-server:netconf-server/listen/endpoint[name='test_ssh_listen_endpt']
[INF] Set item request startup datastore, xpath: /ietf-netconf-server:netconf-server/listen/endpoint[name='test_ssh_listen_endpt']/ssh/address
[INF] Set item request startup datastore, xpath: /ietf-netconf-server:netconf-server/listen/endpoint[name='test_ssh_listen_endpt']/ssh/port
[INF] Set item request startup datastore, xpath: /ietf-netconf-server:netconf-server/listen/endpoint[name='test_ssh_listen_endpt']/ssh/host-keys/host-key[name='test_ssh_listen_key']
[INF] Set item request startup datastore, xpath: /ietf-netconf-server:netconf-server/listen/endpoint[name='test_ssh_listen_endpt']/ssh/host-keys/host-key[name='test_ssh_listen_key']/public-key
[WRN] libyang: Leafref "/ietf-keystore:keystore/ietf-keystore:private-keys/ietf-keystore:private-key/ietf-keystore:name" of value "ssh_host_rsa_key" points to a non-existing leaf. (/ietf-netconf-server:netconf-server/listen/endpoint[name='test_ssh_listen_endpt']/ssh/host-keys/host-key[name='test_ssh_listen_key']/public-key)
[ERR] Data validation failed: Leafref "/ietf-keystore:keystore/ietf-keystore:private-keys/ietf-keystore:private-key/ietf-keystore:name" of value "ssh_host_rsa_key" points to a non-existing leaf.
[ERR] Commit operation failed with 1 error(s).
[ERR] Error returned from sr_commit: Validation of the changes failed.
[ERR] /ietf-netconf-server:netconf-server/listen/endpoint[name='test_ssh_listen_endpt']/ssh/host-keys/host-key[name='test_ssh_listen_key']/public-key : Leafref "/ietf-keystore:keystore/ietf-keystore:private-keys/ietf-keystore:private-key/ietf-keystore:name" of value "ssh_host_rsa_key" points to a non-existing leaf.
Errors were encountered during importing. Cancelling the operation.
[INF] Unlock datastore request
[INF] Dropping session id=341769122.
[INF] Connection Manager stop requested.
[INF] Connection Manager successfully destroyed.
Finished installing libnetconf yang file

**Comment by vishalwc** [comment]
Module Name                   | Revision   | Conformance | Data Owner          | Permissions | Submodules                    | Enabled Features
-----------------------------------------------------------------------------------------------------------------------------------------------
ietf-inet-types               | 2013-07-15 | Installed   |                     |             |                               |
ietf-yang-types               | 2013-07-15 | Installed   |                     |             |                               |
ietf-netconf-monitoring       | 2010-10-04 | Installed   | root:root           | 666         |                               |
ietf-netconf-acm              | 2012-02-22 | Installed   | root:root           | 666         |                               |
ietf-netconf                  | 2011-06-01 | Installed   | root:root           | 666         |                               |
ietf-interfaces               | 2014-05-08 | Installed   | root:root           | 666         |                               | if-mib
iana-if-type                  | 2014-05-08 | Installed   |                     |             |                               |
ietf-netconf-with-defaults    | 2011-06-01 | Installed   |                     |             |                               |
ietf-system                   | 2014-08-06 | Installed   | root:root           | 666         |                               |
iana-crypt-hash               | 2014-08-06 | Installed   |                     |             |                               |
ietf-netconf-server           | 2016-11-02 | Installed   | root:root           | 666         |                               | listen ssh-listen call-home ssh-call-home
ietf-x509-cert-to-name        | 2014-12-10 | Imported    |                     |             |                               |
ietf-ssh-server               | 2016-11-02 | Imported    |                     |             |                               |
ietf-keystore                 | 2016-10-31 | Imported    | root:root           | 666         |                               |
ietf-tls-server               | 2016-11-02 | Imported    |                     |             |                               |
ietf-netconf-notifications    | 2012-02-06 | Installed   | root:root           | 666         |                               |
nc-notifications              | 2008-07-14 | Installed   | root:root           | 666         |                               |
notifications                 | 2008-07-14 | Imported    | root:root           | 666         |                               |

**Comment by vishalwc** [comment]
**Custom script **

rm -f /etc/sysrepo/data/*.lock
rm -f /etc/sysrepo/data/*.startup
rm -f /etc/sysrepo/data/*.running
rm -f /etc/sysrepo/data/*.persist
rm -f /etc/sysrepo/data/internal/*
rm -rf /etc/sysrepo/data/notifications/*

/usr/bin/sysrepoctl --install --yang=/ovn/etc/libnetconf/yang-models/ietf-inet-types@2013-07-15.yang
/usr/bin/sysrepoctl --install --yang=/ovn/etc/libnetconf/yang-models/ietf-yang-types@2013-07-15.yang
/usr/bin/sysrepoctl --install --yang=/ovn/etc/libnetconf/yang-models/ietf-netconf-monitoring@2010-10-04.yang
/usr/bin/sysrepoctl --install --yang=/ovn/etc/libnetconf/yang-models/ietf-netconf-acm@2012-02-22.yang
/usr/bin/sysrepoctl --install --yang=/ovn/etc/libnetconf/yang-models/ietf-netconf@2011-06-01.yang
/usr/bin/sysrepoctl --install --yang=/ovn/etc/libnetconf/yang-models/ietf-interfaces@2014-05-08.yang
/usr/bin/sysrepoctl --install --yang=/ovn/etc/libnetconf/yang-models/iana-if-type.yang
/usr/bin/sysrepoctl --install --yang=/ovn/etc/libnetconf/yang-models/ietf-netconf-with-defaults.yang
/usr/bin/sysrepoctl --install --yang=/ovn/etc/libnetconf/yang-models/ietf-system@2014-08-06.yang
/usr/bin/sysrepoctl --install --yang=/ovn/etc/libnetconf/yang-models/iana-crypt-hash@2014-08-06.yang

/usr/bin/sysrepoctl --install --yang=/ovn/etc/libnetconf/yang-models/ietf-netconf-server@2016-11-02.yang --permission=666
/usr/bin/sysrepoctl --install --yang=/ovn/etc/libnetconf/yang-models/ietf-netconf-notifications@2012-02-06.yang --permissions=666
/usr/bin/sysrepoctl --install --yang=/ovn/etc/libnetconf/yang-models/nc-notifications@2008-07-14.yang --permissions=666

echo "Enable Call-home feature"
/usr/bin/sysrepoctl -m ietf-netconf-server -e listen
/usr/bin/sysrepoctl -m ietf-netconf-server -e ssh-listen
/usr/bin/sysrepoctl -m ietf-netconf-server -e call-home
/usr/bin/sysrepoctl -m ietf-netconf-server -e ssh-call-home
/usr/bin/sysrepoctl -m ietf-interfaces -e if-mib

mkdir -p /usr/local/etc/keystored/keys

cp /etc/ssh/ssh_host_rsa_key /usr/local/etc/keystored/keys/ssh_host_rsa_key.pem
/usr/bin/chmod go-rw /usr/local/etc/keystored/keys/ssh_host_rsa_key.pem
/usr/bin/openssl rsa -pubout -in /usr/local/etc/keystored/keys/ssh_host_rsa_key.pem -out /usr/local/etc/keystored/keys/ssh_host_rsa_key.pub.pem

sysrepocfg -l 3 -d startup -i /etc/sysrepo/yang/stock_key_config.xml ietf-keystore

echo "\nFinished installing keystore\n"
sysrepocfg -l 3 -d startup -i /etc/sysrepo/yang/callhome_bk.xml ietf-netconf-server

echo "Finished installing libnetconf yang file"

**Comment by michalvasko** [comment]
Hi,
I may have discovered the problem, in the output it can be seen that `ietf-keystore` is only `Imported`. That is wrong, it should be `Implemented` after the commands you are executing and I am going to look at this, but it would still be better if you explicitly install this module as well. Coincidentally, it should also fix your problem. So, my suggestion is to add installing `ietf-keystore` before installing `ietf-netconf-server`.

Regards,
Michal

**Comment by vishalwc** [comment]
Thanks Michal for your attention and support

**Comment by michalvasko** [comment]
Hi,
so this problem should be solved by the pull request. If not or you encounter any other issues, please create a new issue, we are already off-topic in this one.

Regards,
Michal
